File: Homeworks/1. Programming/1. C#-Part-1/Other-Solutions/03.Operators-and-Expressions/07. Point in a circle/PointInCircle.cs
/*Problem 7. Point in a Circle
Write a program that reads the coordinates of a point x and y and using an expression checks if given
point (x, y) is inside a circle K({0, 0}, 2) - the center has coordinates (0, 0) and the circle has radius 2.
The program should then print "yes DISTANCE" if the point is inside the circle or "no DISTANCE" if the point is outside the circle.
*/
namespace _07.Point_in_a_circle
{
    using System;

    class PointInCircle
    {
        static void Main()
        {
            double pointX = double.Parse(Console.ReadLine());
            double pointY = double.Parse(Console.ReadLine());
            int radius = 2; // int.Parse(Console.ReadLine());

            IsInCircle(pointX, pointY, radius);
        }

        static void IsInCircle(double pointX, double pointY, int radius)
        {
            //The formula is (x-k)^2 + (y-k)^2 = r^2
            double pointCordinates = Math.Sqrt((pointX * pointX) + (pointY * pointY));
            bool isInside = pointCordinates <= radius;

            Console.WriteLine(isInside ? "yes {0:F2}" : "no {0:F2}", pointCordinates);
        }
    }
}

File: Homeworks/2. Web Design and UI technlogies/3. JavaScript-Part-1/02.Operators and Expressions/06. Point in Circle/PointInCircle.js
/**Problem 6. Point in Circle
 Write an expression that
 checks if given point P(x, y) is within a circle K({0,0}, 2). //{0,0} is the centre and 2 is the radius
 */

/*function CheckInCircle(x, y, radius) {
 //The formula for circle is (px - cx) * (px - cx) + (py - cy) * (py - cy) <= r * r

 var point = (x * x) + (y * y),
 IsInCircle = point <= (radius * radius),
 distance = Math.sqrt(point);

 if (IsInCircle) {
 return console.log('yes ' + distance.toFixed(2));
 } else if (point == 0) {
 return console.log('0');
 }
 else {
 return console.log('no ' + distance.toFixed(2));
 }
 }*/
//CheckInCircle(1, 1.655, 2);

function CheckInCircle(args) {
    //The formula for circle is (px - cx) * (px - cx) + (py - cy) * (py - cy) <= r * r
    var x = +args[0],
        y = +args[1];
    var point = (x * x) + (y * y),
        IsInCircle = point <= (2 * 2),
        distance = Math.sqrt(point);

    if (IsInCircle) {
        console.log('yes ' + distance.toFixed(2));
    } else if (point == 0) {
        return console.log('0');
    }
    else {
       return console.log('no ' + distance.toFixed(2));
    }
}

